src/framework/standard/create_help_command.rs

```diff
@@ -176,40 +183,39 @@ impl CreateHelpCommand {
     #[cfg_attr(feature = "cargo-clippy", allow(useless_if_let_seq))]
     pub(crate) fn finish(self) -> Arc<Help> {
         if self.0.striked_commands_tip == Some(String::new()) {
-            let mut strike_text = String::from("~~`Striked commands`~~ are unavailable because they");
-
-            let mut concat_with_comma = if self.0.lacking_permissions == HelpBehaviour::Strike {
-                let _ = write!(strike_text, " require permissions");
-                true
-            } else {
-                false
-            };
-
-            if self.0.lacking_role == HelpBehaviour::Strike {
-
+            let mut strike_text =
+                String::from("~~`Striked commands`~~ are unavailable because they");
+
+            let reasons = [
+                (self.0.lacking_permissions, " require permissions"),
+                (self.0.lacking_role, " require a specific role"),
+                (self.0.lacking_ownership, " require ownership"),
+                (self.0.wrong_channel, " are limited to DM/guilds"),
+            ];
+            let last_reason = reasons.len() - 1;
+            let mut concat_with_comma = false;
+            for (i, text) in reasons
+                .iter()
+                .filter(|reason| reason.0 == HelpBehaviour::Strike)
+                .map(|reason| reason.1)
+                .enumerate()
+            {
                 if concat_with_comma {
-                    let _ = write!(strike_text, ", require a specific role");
+                    let sep = if i != last_reason { "," } else { " or" };
```

Comment:
`" or"` is not used in the `05_command_framework`-example, probably because `last_reason` is referring to the old length prior `filter()`. Only when setting all `HelpBehaviour`-methods to `Strike` the `" or"` separator is being used.